src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -156,10 +156,22 @@ public void addParticipant(Participant participant) throws AlfredException {
      *
      * @param id
      * @param participant
-     * @return boolean
      */
-    public boolean updateParticipant(Id id, Participant participant) {
-        return this.participantList.update(id, participant);
+    public void updateParticipant(Id id, Participant participant) throws AlfredException {
+        // TODO: Customize error message.
+        try {
+            // Update the participant in the team list as well
+            Team targetTeam = this.getTeamByParticipantId(id);
+            boolean isSuccessful = targetTeam.updateParticipant(participant);
+            if (!isSuccessful) {
+                logger.warning("The participant is not in the team provided");
+                return;
+            }
+
+            this.participantList.update(id, participant);
```

Comment:
And if it fails here, you can throw an exception saying there exists duplicate participants

Reply:
Oh yea Im sending a PR for customized error messages at 3 as Im doing some internal refactoring on model side.